### PR TITLE
feat(learn): add mocha and chai instructions

### PR DESCRIPTION
**Pull request by ShaunSHamilton**
Checklist:

- [x] I have read [freeCodeCamp's contribution guidelines](https://contribute.freecodecamp.org).
- [x] My pull request has a descriptive title (not a vague title like `Update index.md`)
- [x] My pull request targets the `master` branch of freeCodeCamp.
- [x] All the files I changed are in the same world language, for example: only English changes, or only Chinese changes, etc.

Related to: #39720 

Please note: I am aware of how to add new challenges...this was just a _quick-test_

### CODE REVIEW

**Comment by moT01** [comment]
> I am going to need a bit of guidance with how to do the Chai instructions. I will add my PR now, but it will likely need more challenges added

Is there something specific you aren't sure on or want some input on @Sky020? I only took a glance at the diff, at it looks like it's on the right track. I noticed there was a new lesson added - that one was in the boilerplate but not on /learn or something?

**Comment by Sky020** [comment]
>  that one was in the boilerplate but not on /learn or something?

@moT01 The original (and final) lesson in /learn](https://www.freecodecamp.org/learn/quality-assurance/quality-assurance-and-testing-with-chai/run-functional-tests-using-a-headless-browser-ii) basically just says _"now, follow the instructions in the boilerplate"_
> This exercise is similar to the preceding one.
Look at the code for directions. Follow the assertions order: we rely on it.

There are a lot of instructions in the boilerplate, from this section. So, I suspect, I will be adding 4 - 5 lessons. Do you have anything against this?

**Comment by moT01** [comment]
> There are a lot of instructions in the boilerplate, from this section. So, I suspect, I will be adding 4 - 5 lessons. Do you have anything against this?

Not if it's necessary - I haven't gone through it to know if that's the case. I can take a look.

**Comment by moT01** [comment]
I went through it - there's an awful lot of instructions in the boilerplate, not to mention confusion on some of the lessons. Unless I'm missing something, there is a page on /learn for that final lesson - you linked to it - so I don't think that additional lesson you added is necessary?

I'm not sure adding other lessons is necessary either - did you have some in mind that you wanted to explain certain areas better? I feel like the lessons can possibly remain how they are if we can write good instructions on learn for them. I feel like that's what we should shoot for. But maybe if I were to try to do it, I would be in favor of adding more - so I'm not entirely sure.

**Comment by Sky020** [comment]
> I'm not sure adding other lessons is necessary either... I feel like the lessons can possibly remain how they are if we can write good instructions on learn for them.

@moT01 I am very confused by your above comment. Are you saying we should try to fit everything into one lesson?

**Comment by moT01** [comment]
Yes, I am saying that we try and fit everything into the existing lessons and not add more. There would need to be some significant rewriting to accomplish that, but I think it's possible. That's my opinion anyway - but I'm not completely against adding more. I think the overall plan was to not add more lessons since the project based curriculum was announced, but I'm sure an exception could be made if it's necessary.

To try and clarify, and maybe you aren't sure yet - but you want to take that last lesson (the one you linked to), and break it up into several lessons? I'm not sure which specific lesson you are referring to.

Here, I made [a potential example for the second to last lesson](https://gist.github.com/moT01/31c3fb2917a0551dd57c4e4f5da589f7)

**Comment by Sky020** [comment]
@moT01 I have pushed what I thought might work. (I still need to figure something out with the boilerplate)

On a side note: Do you have any idea how I could accomplish what I have done with the ordered-lists, without Husky (linter) complaining?

**Comment by moT01** [comment]
Try just using 1 for all the numbers...
```
1. a 
1. b 
1. c
```

=

1. a 
1. b 
1. c

**Comment by moT01** [comment]
>  I have pushed what I thought might work. 

I actually don't mind breaking that one up - that was the one I thought made the most sense. If we do keep it broken up like that, I would maybe keep the title and mongo id the same for the other two headless lessons and give the new one a new title and mongoid. Maybe "Simulate actions using a Headless Browser". I suppose it works either way, but that's another way I guess.

File: curriculum/challenges/_meta/quality-assurance-and-testing-with-chai/meta.json
Comment (moT01) [inline curriculum/challenges/_meta/quality-assurance-and-testing-with-chai/meta.json:12]:
```suggestion
    [
      "587d824a367417b2b2512c46",
      "Learn How JavaScript Assertions Work"
    ],
```
These could probably all be spaced out for consistency. Probably easier to read on one line like that, but I think all the meta files have them spaced out.

Comment (Sky020) [reply curriculum/challenges/_meta/quality-assurance-and-testing-with-chai/meta.json:12]: I agree, but it was the recommended fCC prettier settings that did that (line too long). So, I am happy to revert it, but should I just add the file to a `.prettierignore` file?

**Review by RandellDawson — CHANGES_REQUESTED** [review]
@Sky020 See my suggested changes.

@Sky020 You will need to look at my suggestions on PR https://github.com/freeCodeCamp/boilerplate-mochachai/pull/27 to understand how these changes will need related to the boilerplate changes  If you have any questions, let me know or contact me on Gitter and we can set up a meeting to discuss the changes further.

**Comment by RandellDawson** [comment]
@Sky020 I sent you a message on Gitter to explain how to add one of the examples to the applicable `/learn` pages.

**Comment by Sky020** [comment]
@RandellDawson The boilerplate is ready. The only thing left to discuss is `Run Functional Tests using a Headless Browser II` - what to do about this example? (how to word it)

**Comment by RandellDawson** [comment]
> what to do about this example? (how to word it)

Just do your best at a first pass.  We can tweak it once you get something in the description section and the comments removed from the example code.

File: curriculum/challenges/english/06-quality-assurance/quality-assurance-and-testing-with-chai/assert-deep-equality-with-.deepequal-and-.notdeepequal.md
Comment (RandellDawson) [inline curriculum/challenges/english/06-quality-assurance/quality-assurance-and-testing-with-chai/assert-deep-equality-with-.deepequal-and-.notdeepequal.md:14]:
```suggestion
`deepEqual()` asserts that two objects are deep equal.
```

File: curriculum/challenges/english/06-quality-assurance/quality-assurance-and-testing-with-chai/compare-the-properties-of-two-elements.md
Comment (RandellDawson) [inline curriculum/challenges/english/06-quality-assurance/quality-assurance-and-testing-with-chai/compare-the-properties-of-two-elements.md:34]:
```suggestion
  - text: You should choose the right assertion - `isAbove` vs. `isAtMost`.
```

**Review by RandellDawson — DISMISSED** [review]
@Sky020 In addition to the changes requested, can you also add backticks around the methods in the following lines of `curriculum/challenges/english/06-quality-assurance/quality-assurance-and-testing-with-chai/learn-how-javascript-assertions-work.md`
- `  - text: You should choose the right assertion - isNull vs. isNotNull.`
- `  - text: You should choose the right assertion - isNull vs. isNotNull.`

Actually, it would be nice if these `text` could explain what they are supposed to be testing.  As it reads right now, they are the exact same text even though the `testString`s are different.  In fact, if you see other such `text` in other challenges of this same section, go ahead and fix those too.

**Comment by Sky020** [comment]
This is failing a Project Euler challenge?!

**Comment by nhcarrigan** [comment]
It looks like the runtime closed - so maybe a timeout issue?

The solution for that Euler challenge passes all the tests on the `/learn` page. Probably best to re-run the CI, IMO.

**Review by moT01 — APPROVED** [review]
I think I'm good with the changes on this PR 🎉 Nice job @Sky020 🎉 

~~I left a comment on the boilerplate PR that needs to be looked into I think.~~

~~Edit: I think we should wait with this one until we figure out the problem with the boilerplate.~~

This should be good to go now pending another review.

**Review by naomi-lgbt — APPROVED** [review]
LGTM~! 🥳 I ran through the new challenges with the updated boilerplate and everything seemed clear enough.

Thank you for your pull request! We look forward to additional contributions!

**Pull request merged**
